packages/xdl/src/UrlUtils.ts

```diff
@@ -11,24 +13,37 @@ import XDLError from './XDLError';
 import ip from './ip';
 import * as ProjectUtils from './project/ProjectUtils';
 
+interface URLOptions extends Omit<ProjectSettings.ProjectSettings, 'urlRandomness'> {
+  urlType: null | 'exp' | 'http' | 'no-protocol' | 'redirect';
+}
+
+interface MetroQueryOptions {
+  dev?: boolean;
+  strict?: boolean;
+  minify?: boolean;
+}
+
 export async function constructBundleUrlAsync(
   projectRoot: string,
-  opts: any,
+  opts: Partial<URLOptions>,
   requestHostname?: string
 ) {
-  return constructUrlAsync(projectRoot, opts, true, requestHostname);
+  return await constructUrlAsync(projectRoot, opts, true, requestHostname);
 }
 
 export async function constructManifestUrlAsync(
   projectRoot: string,
-  opts?: any,
+  opts?: Partial<URLOptions>,
   requestHostname?: string
 ) {
-  return constructUrlAsync(projectRoot, opts, false, requestHostname);
+  return await constructUrlAsync(projectRoot, opts ?? {}, false, requestHostname);
```

Comment:
probably not needed to include `?? {}` here since `constructUrlAsync` supports opts being `null` as well - could extend it to accept `undefined` w/o changing behavior. but doesn't really matter